Ng semantic

bex1 opened the pull request:
Introducing ng-semantic.
https://ng-semantic.herokuapp.com/#/
http://semantic-ui.com/

The register view has been adapted and is utilizing ngSemantic.
It can be used as a template when redesigning other views with forms.

__TODO__

- [x] Make sure `aot` compilation still works
- [x] Move `src/assets/semantic-ui` out of this project and import is as dependency instead
- [x] Update dependencies in `package.json` (ddb7576)
- [x] Decrease size for `vendor.bundle.js` (a329bde), see https://stackoverflow.com/questions/37631098/how-to-bundle-an-angular-2-app-for-production

---

Closes #347
Closes #349
Closes #343
Closes #339
Closes #328 
Closes #330 
Closes #332 
Closes #334 
Closes #345 
Closes #348
Closes #363

buren committed:
Create ApiErrors, ApiError and misc objects

buren committed:
Return ApiErrors instead from api-call.service

buren committed:
Refactor contact.component to use new ApiErrors

bex1 committed:
added ng semantic and reimplemented register

buren committed:
Refactor user-register.component to use new ApiErrors

buren committed:
Refactor user-profile.component to use new ApiErrors

buren committed:
Refactor user-details.component to use new ApiErrors

buren committed:
Remove dead variable from contact.component

buren committed:
Refactor forgot-password.component to use new ApiErrors

buren committed:
Add ApiErrors resetErrorFor method that resets all errors for a given field

buren committed:
Merge branch 'master' of github.com:justarrived/just-match-web into api-errors

bex1 committed:
register component done

bex1 committed:
added client min length validation to zip field

buren committed:
Merge branch 'master' of github.com:justarrived/just-match-web into ng-semantic

buren committed:
nGSemantic

buren committed:
Re-gen yarn.lock

buren committed:
Working ng-semantic

buren committed:
Fix package.json & ng-semantic module import

buren committed:
ngSemantic from custom branch

buren commented on src/assets/semantic-ui/components/label.min.css:
```diff
@@ -0,0 +1,9 @@
+ * Released under the MIT license
+ * http://opensource.org/licenses/MIT
+ *
+ */.ui.label{display:inline-block;line-height:1;vertical-align:baseline;margin:0 .14285714em;background-color:#e8e8e8;background-image:none;padding:.5833em .833em;color:rgba(0,0,0,.6);text-transform:none;font-weight:700;border:0 solid transparent;border-radius:.28571429rem;-webkit-transition:background .1s ease;transition:background .1s ease}.ui.label:first-child{margin-left:0}.ui.label:last-chil ...
```
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
`!important` should not be used.

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
! should be preceded by a space, and should not be followed by a space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#bangformat)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Properties should be ordered position, top, left, z-index, margin

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren replied:
Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 221868 possible new issues (including those that may have been commented here).
* 7 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
Add ng-semantic-ja

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+.ui[class*="pointing above"].label:before {
+  border-width: 1px 0px 0px 1px;
+  -webkit-transform: translateX(-50%) translateY(-50%) rotate(45deg);
+          transform: translateX(-50%) translateY(-50%) rotate(45deg);
```
Line should be indented 2 spaces, but was indented 10 spaces.

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+.ui.pointing.label:before,
+.ui[class*="pointing above"].label:before {
+  border-width: 1px 0px 0px 1px;
+  -webkit-transform: translateX(-50%) translateY(-50%) rotate(45deg);
```
Avoid vendor prefixes.

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#vendorprefix)

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+}
+.ui.pointing.label:before,
+.ui[class*="pointing above"].label:before {
+  border-width: 1px 0px 0px 1px;
```
`0px` should be written without units as `0`.

buren replied:
`0px` should be written without units as `0`.

buren replied:
Properties should be ordered top, left, border-width, transform

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+.ui[class*="pointing above"].label {
+  margin-top: 1em;
+}
+.ui.pointing.label:before,
```
Begin pseudo elements with double colons: `::`

buren replied:
Begin pseudo elements with double colons: `::`

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+.ui.pointing.label,
+.ui[class*="pointing above"].label {
+  margin-top: 1em;
+}
```
Rule declaration should be followed by an empty line

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#emptylinebetweenblocks)

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+  transition: background 0.1s ease;
+}
+
+/*--- Above ---*/
```
Use `//` comments everywhere

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#comment)

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+  width: 0.6666em;
+  height: 0.6666em;
+  -webkit-transition: background 0.1s ease;
+  transition: background 0.1s ease;
```
`0.1` should be written without a leading zero as `.1`

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#leadingzero)

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+  z-index: 2;
+  width: 0.6666em;
+  height: 0.6666em;
+  -webkit-transition: background 0.1s ease;
```
`0.1` should be written without a leading zero as `.1`

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#leadingzero)

buren replied:
Avoid vendor prefixes.

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#vendorprefix)

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+  background-image: none;
+  z-index: 2;
+  width: 0.6666em;
+  height: 0.6666em;
```
`0.6666` should be written without a leading zero as `.6666`

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#leadingzero)

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+          transform: rotate(45deg);
+  background-image: none;
+  z-index: 2;
+  width: 0.6666em;
```
`0.6666` should be written without a leading zero as `.6666`

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#leadingzero)

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+  position: absolute;
+  content: '';
+  -webkit-transform: rotate(45deg);
+          transform: rotate(45deg);
```
Line should be indented 2 spaces, but was indented 10 spaces.

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+.ui.pointing.label:before {
+  position: absolute;
+  content: '';
+  -webkit-transform: rotate(45deg);
```
Avoid vendor prefixes.

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#vendorprefix)

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+
+/* Arrow */
+.ui.pointing.label:before {
+  position: absolute;
```
Properties should be ordered content, position, z-index, width, height, background-image, transition, transform

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+}
+
+/* Arrow */
+.ui.pointing.label:before {
```
Begin pseudo elements with double colons: `::`

buren replied:
Merge rule `.ui.pointing.label:before` with rule on line 1129

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#mergeableselector)

buren commented on src/assets/semantic-ui/components/label.css:
```diff
@@ -0,0 +1,1307 @@
+  border-color: inherit;
+}
+
+/* Arrow */
```
Use `//` comments everywhere

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#comment)

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 221868 possible new issues (including those that may have been commented here).
* 7 fixed issues! 🎉

But beware that this branch is **[13 commits behind the `justarrived:master` branch](https://github.com/justarrived/just-match-web/compare/justarrived:ng-semantic...justarrived:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
Add semantic-ui as a dependency

buren committed:
Remove src/assets/semantic-ui/

buren commented on src/app/views/user/user-register/user-register.component.scss:
```diff
@@ -1,72 +1,22 @@
-    top: 5px;
-    color: $white;
+  .save-message {
+    color: green;
```
Color `green` should be written in hexadecimal form as `#008000`

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#colorkeyword)

buren commented on src/app/views/user/user-register/user-register.component.scss:
```diff
@@ -1,72 +1,22 @@
+    text-align: center;
+
+    .register-form-button {
+      margin: 15px auto 30px;
```
Properties should be ordered min-width, margin

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/app/views/user/user-register/user-register.component.scss:
```diff
@@ -1,72 +1,22 @@
-        color: $color-cerise-500;
-      }
+  .register-form-button-container {
+    margin-top: 20px;
```
Properties should be ordered padding-bottom, margin-top, text-align

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 7 fixed issues! 🎉

But beware that this branch is **[13 commits behind the `justarrived:master` branch](https://github.com/justarrived/just-match-web/compare/justarrived:ng-semantic...justarrived:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
Add translations to initial loading screen

buren committed:
Upgrade angular-cli, zone.js, rxjs, ts-node, @angular/compiler-cli and others

buren commented on src/app/styles/typography.scss:
```diff
@@ -27,19 +27,19 @@ $fs-xxxl: 32px;
 @font-face {
     font-family: 'museo';
-    src: url('./assets/fonts/museo/museo-700.otf') format("opentype");
+    src: url('/assets/fonts/museo/museo-700.otf') format("opentype");
```
Prefer single quoted strings

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#stringquotes)

buren replied:
Line should be indented 2 spaces, but was indented 4 spaces.

buren commented on src/app/styles/typography.scss:
```diff
@@ -27,19 +27,19 @@ $fs-xxxl: 32px;
 @font-face {
     font-family: 'museo';
-    src: url('./assets/fonts/museo/museo-500.otf') format("opentype");
+    src: url('/assets/fonts/museo/museo-500.otf') format("opentype");
```
Prefer single quoted strings

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#stringquotes)

buren replied:
Line should be indented 2 spaces, but was indented 4 spaces.

buren commented on src/app/styles/typography.scss:
```diff
@@ -27,19 +27,19 @@ $fs-xxxl: 32px;
 @font-face {
   font-family: 'museo';
-  src: url('./assets/fonts/museo/museo-300.otf') format("opentype");
+  src: url('/assets/fonts/museo/museo-300.otf') format("opentype");
```
Prefer single quoted strings

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#stringquotes)

buren commented on protractor.conf.js:
```diff
@@ -2,7 +2,7 @@
 
 /*global jasmine */
-var SpecReporter = require('jasmine-spec-reporter');
+const { SpecReporter } = require('jasmine-spec-reporter');
```
Possible undefined variable: 'require' is not defined.

If this is intended to be a global variable, you can supress this warning by adding a  `/*global [VARIABLE NAME] */` comment on this file or set it under the `globals` section of your `.eslintrc` configuration.

[Read more about it here.](http://eslint.org/docs/rules/no-undef)

buren committed:
Explicitly define the latest version of ng2 in package.json

buren committed:
Upgrade @angular/router and generate new yarn.lock

buren committed:
Update extract-text-webpack-plugin

buren committed:
Configure @angular/router to perform intialNavigation

buren commented:
Tried to upgrade to the latest `@types/lodash` but got this stack trace when trying to build the app:

```
ERROR in ~/app/src/app/services/system-languages.service.ts (3,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'orderBy'.)
ERROR in ~/app/src/app/services/proxy/user-proxy.service.ts (5,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'map'.)
ERROR in ~/app/src/app/services/proxy/country-proxy.service.ts (4,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'map'.)
ERROR in ~/app/src/app/services/proxy/language-proxy.service.ts (4,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'map'.)
ERROR in ~/app/src/app/services/proxy/skill-proxy.service.ts (3,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'map'.)
ERROR in ~/app/src/app/views/home/home.component.ts (10,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'isEmpty'.)
ERROR in ~/app/src/app/views/my-jobs/user-jobs/user-jobs.component.ts (5,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'map'.)
ERROR in ~/app/src/app/components/autocomplete-dropdown/autocomplete-dropdown.component.ts (2,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'cloneDeep'.)
~/app/src/app/components/autocomplete-dropdown/autocomplete-dropdown.component.ts (2,20): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'some'.)
~/app/src/app/components/autocomplete-dropdown/autocomplete-dropdown.component.ts (2,26): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'isEqual'.)
~/app/src/app/components/autocomplete-dropdown/autocomplete-dropdown.component.ts (2,35): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'isObject'.)
~/app/src/app/components/autocomplete-dropdown/autocomplete-dropdown.component.ts (2,45): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'assignIn'.)
~/app/src/app/components/autocomplete-dropdown/autocomplete-dropdown.component.ts (2,55): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'filter'.)
ERROR in ~/app/src/app/components/comments/comments.component.ts (6,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'orderBy'.)
ERROR in ~/app/src/app/views/user/user-settings/user-profile/user-profile.component.ts (14,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'isEmpty'.)
~/app/src/app/views/user/user-settings/user-profile/user-profile.component.ts (14,18): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'some'.)
~/app/src/app/views/user/user-settings/user-profile/user-profile.component.ts (14,24): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'map'.)
ERROR in ~/app/src/app/models/user.ts (2,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'map'.)
~/app/src/app/models/user.ts (2,14): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'find'.)
ERROR in ~/app/src/app/services/proxy/job-proxy.service.ts (3,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'map'.)
ERROR in ~/app/src/app/services/proxy/faq-proxy.service.ts (4,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'map'.)
ERROR in ~/app/src/app/services/proxy/comments-proxy.service.ts (4,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'map'.)
ERROR in ~/app/src/app/models/user/user-language.ts (4,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'find'.)
ERROR in ~/app/src/app/models/user/user-skill.ts (4,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'find'.)
ERROR in ~/app/src/app/models/job/job.ts (3,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'map'.)
ERROR in ~/app/src/app/models/company.ts (1,9): Module '"~/app/node_modules/@types/lodash/index"' has no exported member 'find'.)
```

buren committed:
Upgrade packages to latest version: @types/node @types/lodash @types/jasmine webpack

buren committed:
Merge branch 'api-errors' of github.com:justarrived/just-match-web into ng-semantic

buren commented on src/app/services/api-call.service.ts:
```diff
@@ -110,6 +108,7 @@ export class ApiCall {
     return params;
   }
 
+  // TODO: Add typing.. currently returns ErrorObservable<ApiErrors> (though I can't seem to import that symbol from rxjs...)
```
TODO found

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).
* 7 fixed issues! 🎉

But beware that this branch is **[13 commits behind the `justarrived:master` branch](https://github.com/justarrived/just-match-web/compare/justarrived:ng-semantic...justarrived:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
Merge branch 'master' of github.com:justarrived/just-match-web into ng-semantic

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).
* 7 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
:hocho: double-quotes in typography.scss

buren commented on src/app/styles/typography.scss:
```diff
@@ -27,24 +27,24 @@ $fs-xxxl: 32px;
 @font-face {
     font-family: 'museo';
-    src: url('./assets/fonts/museo/museo-700.otf') format("opentype");
+    src: url('/assets/fonts/museo/museo-700.otf') format('opentype');
```
Line should be indented 2 spaces, but was indented 4 spaces.

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).
* 8 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
Add user-register.component form loading I18n

buren committed:
:hocho: serverValidation reference in user-profile.component

buren committed:
Update user-details.component to user Semantic UI

buren committed:
Add api-errors refactor to user-profile.component

bex1 committed:
Reset api url in environment.ts

buren commented:
Hello, @bex1! This is your first Pull Request that will be reviewed by [Ebert](https://ebertapp.io/), an automatic Code Review service. It will leave comments on this diff with potential issues and style violations found in the code as you push new commits. You can also see all the issues found on this Pull Request on its [review page](https://ebertapp.io/github/justarrived/just-match-web/pulls/356). Please check our [documentation](https://ebertapp.io/docs) for more information.

bex1 committed:
Change to semantic-ui-css as dependency

buren committed:
Rename loading-gif/ => loading/

buren committed:
Create ApiErrorsComponent and refactor all forms to use it

bex1 committed:
typo in package.json

buren committed:
Convert contact.component to Semantic UI

buren committed:
:hocho: dead contact.component SASS

buren committed:
Split contact.component into contact-page.component and contact-form.component

buren committed:
Remove unneeded imports in contact-page.component

buren committed:
Remove contact-form.component from start page

buren committed:
Merge branch 'ng-semantic' of github.com:justarrived/just-match-web into ng-semantic

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 11 possible new issues (including those that may have been commented here).
* 16 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

bex1 committed:
Added data fields to register slect inputs. Just to show how preselect works now. (In register there is no preselected value however ofc)

bex1 committed:
Merge branch 'ng-semantic' of https://github.com/justarrived/just-match-web into ng-semantic

buren committed:
Working input-errors.component

buren committed:
Add ability to display API errors in input-errors.component

buren committed:
Refactor user-register.component to user input-errors.component

buren committed:
Merge branch 'ng-semantic' of github.com:justarrived/just-match-web into ng-semantic

buren committed:
Refactor user-profile.component to user input-errors.component

buren committed:
Refactor user-details.component to user input-errors.component

buren committed:
:hocho: dead translations strings

buren committed:
Template :lipstick: & :hocho: dead translation

buren committed:
SASS :lipstick:

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 7 possible new issues (including those that may have been commented here).
* 16 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 11 possible new issues (including those that may have been commented here).
* 16 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 7 possible new issues (including those that may have been commented here).
* 16 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
Add name field to User model and refactor contact-form.component

buren commented on src/app/components/loading/loading.component.scss:
```diff
@@ -0,0 +1,5 @@
+.loading {
+  margin: 0 auto;
+  opacity: .7;
```
Properties should be ordered margin, text-align, opacity

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren committed:
Use User name where applicable

bex1 committed:
Fix so contact view doesn't crash when not logged in.

buren committed:
Convert part of comments.component to use the grid from Semantic UI

buren committed:
Merge branch 'ng-semantic' of github.com:justarrived/just-match-web into ng-semantic

buren committed:
Refactor input-errors.component params

buren committed:
Convert login page to Semantic UI. Closes #347

buren committed:
Add support for API errors to login.component

buren committed:
Convert comments.component to Semantic UI. Closes #328

buren committed:
Tweak comments.component grid

buren committed:
Convert job-list-item.component to Semantic UI

buren committed:
SASS :lipstick:

buren commented on src/app/components/job-list-item/job-list-item.component.scss:
```diff
@@ -1,33 +1,34 @@
-  margin: 0 auto;
-  cursor: pointer;
   cursor: hand;
+  cursor: pointer;
```
Property `cursor` already defined on line 12

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#duplicateproperty)

buren commented on src/app/components/job-list-item/job-list-item.component.scss:
```diff
@@ -1,33 +1,34 @@
 }
 
 .job-list-container {
+  border-bottom: 1px solid $color-gray-50;
```
Properties should be ordered max-width, padding, margin, color, cursor, border-bottom

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 10 possible new issues (including those that may have been commented here).
* 18 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
Convert job-preview.component to Semantic UI. Closes #332

buren committed:
Convert pager.component to Semantic UI. Closes #334

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 10 possible new issues (including those that may have been commented here).
* 19 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
:hocho: dead scss

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 10 possible new issues (including those that may have been commented here).
* 22 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
Convert job-details.component to Semantic UI. Closes #345

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 10 possible new issues (including those that may have been commented here).
* 23 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
Convert my-jobs.component to Semantic UI. Closes #348

buren committed:
Convert user-settings.component to Semantic UI. Closes #363

buren commented on src/app/views/user/user-settings/user-settings.component.scss:
```diff
@@ -39,68 +38,69 @@
       }
 
       .user-profile-subheader-info-error-msg {
+        color: $color-cerise-500;
```
Properties should be ordered width, margin-top, color

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/app/views/user/user-settings/user-settings.component.scss:
```diff
@@ -39,68 +38,69 @@
       }
 
       .user-profile-subheader-info-success-msg {
+        color: green;
```
Color `green` should be written in hexadecimal form as `#008000`

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#colorkeyword)

buren replied:
Properties should be ordered width, margin-top, color

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/app/views/user/user-settings/user-settings.component.scss:
```diff
@@ -39,68 +38,69 @@
+          width: 85px;
 
           .fa-camera {
+            color: rgba(255, 255, 255, 0.7);
```
`0.7` should be written without a leading zero as `.7`

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#leadingzero)

buren replied:
Properties should be ordered display, font-size, color, vertical-align

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/app/views/user/user-settings/user-settings.component.scss:
```diff
@@ -39,68 +38,69 @@
 
         .icon-camera {
-          width: 85px;
+          bottom: 0;
```
Properties should be ordered position, top, right, bottom, left, display, width, height, pointer-events

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/app/views/user/user-settings/user-settings.component.scss:
```diff
@@ -39,68 +38,69 @@
 
         .file {
-          width: 85px;
+          cursor: pointer;
```
Properties should be ordered width, height, cursor, opacity

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 17 possible new issues (including those that may have been commented here).
* 31 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

bex1 committed:
HTML alignment convention

bex1 committed:
Refactor user-register.component according to HTML conventions

bex1 committed:
Remove dead register component ts code

bex1 committed:
Add ts import conventions

bex1 committed:
Use aync pipes to handle proxy promises in register component.

bex1 committed:
Added ts constructor conventions

bex1 committed:
Update doc angular and ts versions

bellevik committed:
Pinned Angular router at 3.3.0

Latest version of Angular router is incompatible with other Angular 2
dependencies.

buren commented:
Hello, @bex1! This is your first Pull Request that will be reviewed by [Ebert](https://ebertapp.io/), an automatic Code Review service. It will leave comments on this diff with potential issues and style violations found in the code as you push new commits. You can also see all the issues found on this Pull Request on its [review page](https://ebertapp.io/github/justarrived/just-match-web/pulls/356). Please check our [documentation](https://ebertapp.io/docs) for more information.

bex1 committed:
Move contact form component to forms/contact-form and apply conventions

bex1 committed:
Merge branch 'ng-semantic' of https://github.com/justarrived/just-match-web into ng-semantic

bellevik committed:
Added yarn.lock to .gitignore

bex1 committed:
Added component initialization convention to docs

bex1 committed:
Merge branch 'ng-semantic' of https://github.com/justarrived/just-match-web into ng-semantic

bex1 committed:
refactor register form to a component and make all providers singleton

buren commented on src/app/components/forms/register-form/register-form.component.scss:
```diff
@@ -0,0 +1,20 @@
+}
+
+.save-message {
+  color: green;
```
Color `green` should be written in hexadecimal form as `#008000`

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#colorkeyword)

buren commented on src/app/components/forms/register-form/register-form.component.scss:
```diff
@@ -0,0 +1,20 @@
+  text-align: center;
+
+  .register-form-button {
+    margin: 15px auto 30px;
```
Properties should be ordered min-width, margin

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/app/components/forms/register-form/register-form.component.scss:
```diff
@@ -0,0 +1,20 @@
+@import '../../../styles/colors';
+
+.register-form-button-container {
+  margin-top: 20px;
```
Properties should be ordered padding-bottom, margin-top, text-align

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 20 possible new issues (including those that may have been commented here).
* 34 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

bex1 committed:
Relaxed OnInit implement requirement in conventions

bex1 committed:
Reset apierror on valuechange

bex1 committed:
remove ngclass error on register form fields

bex1 committed:
Contact form changes

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 19 possible new issues (including those that may have been commented here).
* 34 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

bex1 committed:
Added inline vs file html template conventions

buren commented on src/app/components/forms/contact-form/contact-form.component.scss:
```diff
@@ -0,0 +1,10 @@
+  width: 80%;
+
+  .contact-button {
+    margin: 0;
```
Properties should be ordered width, margin

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/app/components/forms/contact-form/contact-form.component.scss:
```diff
@@ -0,0 +1,10 @@
+.contact-button-container {
+  margin: 0 auto;
```
Properties should be ordered width, padding-bottom, margin

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

bex1 committed:
Line length conventions

buren committed:
Configure production build

buren committed:
Update package.json scripts

bex1 committed:
add initial tslint. needs config

bex1 committed:
Merge branch 'ng-semantic' of https://github.com/justarrived/just-match-web into ng-semantic

bex1 committed:
lint fixes

bex1 committed:
lint fixes

bex1 committed:
reset lint command

bex1 committed:
Update line length to 120

bellevik committed:
Removed tracking of yarn-lock

bex1 committed:
Rename user-register to register-page

bex1 committed:
Merge branch 'ng-semantic' of https://github.com/justarrived/just-match-web into ng-semantic

bellevik committed:
Made forgot-password form a component

buren commented on src/app/components/forms/forgot-password-form/forgot-password-form.component.scss:
```diff
@@ -0,0 +1,20 @@
+  }
+
+  .forgot-password-form-link {
+    color: $color-accent-main;
```
Properties should be ordered font-weight, color

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/app/components/forms/forgot-password-form/forgot-password-form.component.scss:
```diff
@@ -0,0 +1,20 @@
+
+  .forgot-password-form-button {
+    display: block;
+    margin: 10px auto;
```
Properties should be ordered display, min-width, margin

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 21 possible new issues (including those that may have been commented here).
* 38 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

Ekman1991 committed:
Extracted reset-password to component

Ekman1991 committed:
Merge remote-tracking branch 'origin/ng-semantic' into ng-semantic

# Conflicts:
#	src/app/app.module.ts
#	src/app/app.routing.module.ts

buren commented on src/app/components/forms/reset-password-form/reset-password-form.component.scss:
```diff
@@ -0,0 +1,15 @@
+
+  .reset-password-form-button {
+    display: block;
+    margin: 10px auto;
```
Properties should be ordered display, min-width, margin

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 22 possible new issues (including those that may have been commented here).
* 42 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

bex1 committed:
refactor login to ng semantic

bex1 committed:
Merge branch 'ng-semantic' of https://github.com/justarrived/just-match-web into ng-semantic

buren commented on src/app/components/forms/login-form/login-form.component.scss:
```diff
@@ -0,0 +1,23 @@
+  }
+
+  .login-form-link {
+    color: $color-accent-main;
```
Properties should be ordered display, padding-bottom, font-weight, color

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/app/components/forms/login-form/login-form.component.scss:
```diff
@@ -0,0 +1,23 @@
+  }
+
+  .login-form-button {
+    margin: 10px auto;
```
Properties should be ordered min-width, margin

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/app/components/forms/login-form/login-form.component.scss:
```diff
@@ -0,0 +1,23 @@
+
+
+.login-form-button-container {
+  text-align: center;
```
Properties should be ordered margin-top, text-align

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 25 possible new issues (including those that may have been commented here).
* 47 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

bex1 committed:
inline register page html

bex1 committed:
one_time_token api error handler on change password

Ekman1991 committed:
Extracted user-details-form to component

buren commented on src/app/components/forms/user-details-form/user-details-form.component.scss:
```diff
@@ -0,0 +1,20 @@
+
+  .user-details-form-button {
+    display: block;
+    margin: 15px auto 30px;
```
Properties should be ordered display, min-width, margin

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

buren commented on src/app/components/forms/user-details-form/user-details-form.component.scss:
```diff
@@ -0,0 +1,20 @@
+  text-align: center;
+
+  .user-details-form-save-message {
+    color: green;
```
Color `green` should be written in hexadecimal form as `#008000`

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#colorkeyword)

buren commented:
Ebert has finished reviewing this Pull Request and has found:

* 27 possible new issues (including those that may have been commented here).
* 53 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/justarrived/just-match-web/pulls/356.

<!-- Ebert auto generated comment -->

buren committed:
Fix api-errors for JSONAPI errors with no source key

buren committed:
Add gzip compression to node server

buren committed:
Add express compression middleware before static assets

buren committed:
Merge branch 'master' of github.com:justarrived/just-match-web into ng-semantic
